### ISSUE TITLE
MLKR optimization algorithm does not work as expected regarding iterations

### ISSUE THREAD

**Issue by wdevazelhes** [metric-learn/metric-learn#104]
`MLKR` does not have the expected behaviour regarding iterations. Indeed, it returns always 0 iterations. (Setting the `max_iter` does not seem to change this). 

### Minimal Example
```python
from sklearn.datasets import make_regression
from metric_learn import MLKR

X, y = make_regression()
mlkr = MLKR(max_iter=5)
mlkr.fit(X, y)
print(mlkr.n_iter_)
```

### Expected result: 

`5` (or between 1 and 5)

### Actual result: 

`0`

### Environment
Linux-4.4.0-130-generic-x86_64-with-debian-stretch-sid
Python 3.4.5 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:47:47) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
NumPy 1.15.0
SciPy 1.1.0
Scikit-Learn 0.19.2
Metric-Learn 0.4.0 (master (last commit: https://github.com/metric-learn/metric-learn/commit/faa240fd7469176036a91430ae6a0a45e627c94a))

**Comment by perimosocordiae** [comment]
Strange. We just forward it on to scipy's `minimize` function, so I'm not sure what's happening.

**Comment by wdevazelhes** [comment]
Turns out in the previous example the optimization result said it converged at first iteration, but trying with this example the behaviour is as expected:

```python
In [1]: 
   ...: import numpy as np
   ...: from metric_learn import MLKR
   ...: np.random.seed(0)
   ...: 
   ...: X = np.random.randn(5, 2)
   ...: y = np.random.randn(5, 1)
   ...: 
   ...: mlkr = MLKR(max_iter=3)
   ...: mlkr.fit(X, y)
   ...: print(mlkr.n_iter_)
   ...: 

```

Returns: 
`3`

Converging at first iteration in the previous example is still weird but should be due to numerical problems (see PR #111 comment: https://github.com/metric-learn/metric-learn/pull/111/comment#issuecomment-412071625)

Closing, since the real problem is tackled in PR #111
